diff --git a/node_modules/react-native-calendars/src/agenda/index.js b/node_modules/react-native-calendars/src/agenda/index.js
index f5946ae..901527f 100644
--- a/node_modules/react-native-calendars/src/agenda/index.js
+++ b/node_modules/react-native-calendars/src/agenda/index.js
@@ -185,6 +185,7 @@ export default class Agenda extends Component {
     onLayout = (event) => {
         this.viewHeight = event.nativeEvent.layout.height;
         this.viewWidth = event.nativeEvent.layout.width;
+        this.calendar?.current?.scrollToDay(this.state.selectedDay.clone(), this.calendarOffset(), false);
         this.forceUpdate();
     };
     onTouchStart = () => {
diff --git a/node_modules/react-native-calendars/src/agenda/reservation-list/reservation.js b/node_modules/react-native-calendars/src/agenda/reservation-list/reservation.js
index a547421..b11c9d6 100644
--- a/node_modules/react-native-calendars/src/agenda/reservation-list/reservation.js
+++ b/node_modules/react-native-calendars/src/agenda/reservation-list/reservation.js
@@ -23,32 +23,32 @@ class Reservation extends Component {
         super(props);
         this.style = styleConstructor(props.theme);
     }
-    shouldComponentUpdate(nextProps) {
-        const d1 = this.props.date;
-        const d2 = nextProps.date;
-        const r1 = this.props.item;
-        const r2 = nextProps.item;
-        let changed = true;
-        if (!d1 && !d2) {
-            changed = false;
-        }
-        else if (d1 && d2) {
-            if (d1.getTime() !== d2.getTime()) {
-                changed = true;
-            }
-            else if (!r1 && !r2) {
-                changed = false;
-            }
-            else if (r1 && r2) {
-                if ((!d1 && !d2) || (d1 && d2)) {
-                    if (isFunction(this.props.rowHasChanged)) {
-                        changed = this.props.rowHasChanged(r1, r2);
-                    }
-                }
-            }
-        }
-        return changed;
-    }
+    // shouldComponentUpdate(nextProps) {
+    //     const d1 = this.props.date;
+    //     const d2 = nextProps.date;
+    //     const r1 = this.props.item;
+    //     const r2 = nextProps.item;
+    //     let changed = true;
+    //     if (!d1 && !d2) {
+    //         changed = false;
+    //     }
+    //     else if (d1 && d2) {
+    //         if (d1.getTime() !== d2.getTime()) {
+    //             changed = true;
+    //         }
+    //         else if (!r1 && !r2) {
+    //             changed = false;
+    //         }
+    //         else if (r1 && r2) {
+    //             if ((!d1 && !d2) || (d1 && d2)) {
+    //                 if (isFunction(this.props.rowHasChanged)) {
+    //                     changed = this.props.rowHasChanged(r1, r2);
+    //                 }
+    //             }
+    //         }
+    //     }
+    //     return changed;
+    // }
     renderDate(date, item) {
         if (isFunction(this.props.renderDay)) {
             return this.props.renderDay(date, item);
